Added configuration example for the BQ Hephestos 2 3D printer.

jbrazio opened the pull request:
Added support for BQ ZUM Mega 3D board and stock Hephestos 2 thermistor. I've taken the liberty to choose the IDs for the board and the thermistor, if they are not inline with upstream drop a comment and I'll fix them.

jbrazio committed:
Added configuration example for the BQ Hephestos 2 3D printer.
Added support for BQ ZUM Mega 3D board and stock Hephestos 2 thermistor.

thinkyhead commented on Marlin/Conditionals.h:
```diff
@@ -45,11 +45,27 @@
 
+  #if ENABLED(BQ_LCD_SMART_CONTROLLER)
+    #define REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER
+    #define SDSUPPORT
```
`SDSUPPORT` is now an option that users must enable explicitly. It was changed so that it can simply be left disabled to turn `SDSUPPORT` off. So you should leave it out here, but enable it in the `Hephestos_2/Configuration.h` file.

jbrazio replied:
I guess people may use the ZUM 3D board and not the LCD module.
Fixed.

thinkyhead commented on Marlin/boards.h:
```diff
@@ -61,6 +61,7 @@
 #define BOARD_MKS_BASE          40   // MKS BASE 1.0
 #define BOARD_BAM_DICE          401  // 2PrintBeta BAM&DICE with STK drivers
 #define BOARD_BAM_DICE_DUE      402  // 2PrintBeta BAM&DICE Due with STK drivers
+#define BOARD_BQ_ZUM_MEGA_3D	503  // bq ZUM Mega 3D
```
You should use spaces here, no tabs. The number 503 is fine, unless there's a board that this is _really_ close to.

jbrazio replied:
The tab issue is gone.

jbrazio committed:
Applying suggestions by @thinkyhead at MarlinFirmware/Marlin#3049.
 - Remove the forced activation of SDSUPPORT, I guess people may use the ZUM
3D board and not the LCD module.
 - Fixed the tab issue within boards.h file.

thinkyhead commented:
I noted that most of the pin assignments are the same as RAMPS 1.3, so I've re-implemented the pins file as overrides to `pins_RAMPS_13.h`. Check out the commit here: https://github.com/MarlinFirmware/Marlin/commit/a89d43c2dce3e87d3a67d5b8c54aa699c3fd245a

Note: The tweaks I made related to `DIGIPOTSS_PIN` just remove the need for `#undef DIGIPOTSS_PIN` in the pins files.

jbrazio commented:
Actually I did based my pins file on RAMPS 1.3 but as the board was from another manufacturer I went to a stand alone definition but this is a much more cleaner solution. Just for QA I'll merge this commit and give it a spin on my bqh2.

jbrazio committed:
Merge remote-tracking branch 'upstream/rcbugfix' into bqh2-support

jbrazio committed:
Merge remote-tracking branch 'upstream/rcbugfix' into bqh2-support

jbrazio committed:
Fine tuned XYJERK.
Disallowed the Z_PROBE_OFFSET_RANGE_MAX to be greater than zero.

thinkyhead committed:
Implemented BQ ZUM Mega 3D board layout as an override to RAMPS 1.3

jbrazio committed:
Added readme file for bqh2 and disabled the SLOWDOWN config flag

thinkyhead commented:
Great. If it all works for you, I don't mind merging this with `RCBugFix`. This will also need to be submitted to `dev` – but take heed! There are some additional requirements for custom configurations there.

jbrazio commented:
@thinkyhead you may merge it.

jbrazio committed:
Merge remote-tracking branch 'upstream/rcbugfix' into bqh2-support

jbrazio committed:
Replaced stock BQ PID values with fine tuned ones generated using M303

jbrazio commented:
This configuration will work with any stock Hephestos 2 printer and provide full support to all the hardware features. @thinkyhead on dev it should be considered a "custom" config or a "vendor" config ?

thinkyhead commented:
> on dev it should be considered a "custom" config or a "vendor" config?

I don't know, but I will ask around. (answering your email at the moment...)

jbrazio commented:
Thanks

CONSULitAS commented:
@jbrazio @thinkyhead 

Look at https://github.com/MarlinFirmware/MarlinDev/blob/dev/configurations/vendor/OpenBeam/KosselPro/Customization.h and https://github.com/MarlinFirmware/MarlinDev/blob/dev/configurations/custom/my_kossel_pro/Customization.h

this is included in https://github.com/MarlinFirmware/MarlinDev/blob/6d0bb88adf66f29526f6028d7208fc10df6a5432/IncludeUserCustomization.h

You will see, that the custom AND the vendor file will be included. The custom is just a delta to the vendor.

So if it is the full config: vendor
And if it is the delta config: custom

Or, what do you think?

jbrazio commented:
Now that you explain the concept it's ok, but looking at the the folder structure I had the impression that Marlin would stop shipping "example" configs and start with "validated" ones under the "vendor" folder thus making "custom" the place for the users to put their configs.

So this config for the bqh2 should go into the vendor as it is a full model.

thinkyhead commented:
> Or, what do you think?

I must plead total ignorance of the new versioning scheme. I believe the developer responsible is also on vacation. I will be able to judge the PR better once I have a grasp of it.

jbrazio commented:
@thinkyhead but should we merge this PR into rcbugfix and I open a new PR for dev to proceed with this discussion ?

thinkyhead commented:
I'm in the process of pushing through PRs for RC4. I'm merging this now, and then I may have to update your configuration files with a few adjustments that have been made from PRs merged earlier.

jbrazio commented:
From what I saw in the merge for this PR the board inheritance from RAMPS didn't went trough, we'll fix that when things go stable.

Good luck with the merge ! :-)
